google/cloud/security/common/gcp_api/_base_client.py

```diff
@@ -69,3 +71,32 @@ def _execute(self, request):
             upstream.
         """
         return request.execute()
+
+    def _build_paged_result(self, request, api_stub, rate_limiter):
+        """Execute results and page through the results.
+
+        Args:
+            request: GCP API client request object.
+            api_stub: The API stub used to build the request.
+            rate_limiter: An instance of RateLimiter to use.
+
+        Returns:
+            A list of API response objects (dict).
+
+        Raises:
+            When the retry is exceeded, exception will be thrown.  This
+            exception is not wrapped by the retry library, and will be handled
+            upstream.
+        """
+        results = []
+        response = []
+        while request is not None:
+            try:
+                with rate_limiter:
+                    response.append(self._execute(request))
+                    results.extend(response)
+                    request = api_stub.list_next(request, response)
```

Comment:
nit: It would be nice if we can add a comment here or in the method doc, to note that user should verify that list_next() does exist for the api being used.